### DISCUSSION TITLE
Lifecycle hooks in functional effects, how to use it?

### DISCUSSION THREAD

**Discussion by thedevtoni** [ngrx/platform#3905] [Q&A]
When migrating my class-based effects to functional effects, I'm trying to make the effects work with the lifecycle hooks, but somehow this is not working. is there a way to use lifecycle hooks with functional effects?

Example:

```typescript
export const init2 = createEffect(
  (actions = inject(Actions)) => {
    return actions.pipe(
      ofType(ROOT_EFFECTS_INIT),
      tap(console.log)
    );
  },
  {
    functional: true,
    dispatch: false,
  }
);```

**Comment by pgg** [comment]
hi @thedevtoni , i have also been looking at this, and have not seen anything official. i have been using the ENVIRONMENT_INITIALIZER for feature effects init, i.e:
```
{
    provide: ENVIRONMENT_INITIALIZER,
    multi: true,
    useValue: () => inject(Store<FeaturePartialState>).dispatch(FeatureActions.load()),
  }
```

**Reply by thedevtoni** [reply]
Top. That seems to work. Thanks.
